addon/mixins/attachable.js

```diff
@@ -16,7 +16,7 @@ export default Ember.Mixin.create({
     var adapter, attachmentKey, data, formData, promise, root, serializer, url,
     _this = this;
     adapter = this.store.adapterFor(this.constructor);
-    serializer = this.store.serializerFor(this.constructor.typeKey);
+    serializer = this.store.serializerFor(this._modelName);
```

Comment:
@mbutsko You're not calling the `_modelName` function anywhere. You're missing the `()` in front of the function name. Like this: `this._modelName()`. This is why all tests are failing.